test/test.js

```diff
@@ -5,16 +5,53 @@ import { test } from '../src/index.js';
 describe('test', function () {
   it('should detect as path template', function () {
     assert.isTrue(test('/{path}'));
+    // trailing slash is allowed
+    assert.isTrue(test('/{path}/'));
     assert.isTrue(test('/pets/{petId}'));
     assert.isTrue(test('/{entity}/me'));
     assert.isTrue(test('/books/{id}'));
     assert.isTrue(test('/a{test}'));
+    assert.isTrue(test('/{test}a'));
+    // parentheses are allowed
+    assert.isTrue(test('/range({x},{y})'));
+    // repeated parameter names are allowed
+    assert.isTrue(test('/range({x},{y})/secondRange({x},{y})'));
     assert.isTrue(test('/{entity}/{another-entity}/me'));
+    // NOTE: the following scenario should probably be considered invalid, query parameters are not allowed in path templates
```

Comment:
I **agree**. This should **not** be considered valid `/{path}` as defined by OpenAPI 3.x.y specification. Specification does define `/{path}` as `A relative path to an individual endpoint.`. 

In https://github.com/OAI/OpenAPI-Specification/issues/3256#issue-1681734800 it was mentioned, that there should be no restrictions where template variables are used.

My interpretation of this was to understand `/{path}` as [Relative Reference](https://datatracker.ietf.org/doc/html/rfc3986#section-4.2) and not as [Path](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3). The reason is that I've seen people defining path templates as follows instead of using Parameter Object (query type):

- `pets?category=3`
- `pets?category=4`

Although in 1.0 version of this library I've put a limitation in, [not to allow template expression](https://github.com/OAI/OpenAPI-Specification/issues/3256#issuecomment-2102558551) in query part of the Relative Reference. `pets?category={category}` will not pass the test.

In 2.0 version of this library we go for this breaking change.

Reply:
I didn't realize this was published as a package and consumed by a lot of people! This is great, the ABNF you've built already got a lot of "real world" exposure.
Yes, following semver here will help to avoid surprising people.

Fixed query parameters, the constant value scenario is interesting. Couldn't a const schema be used for the query parameter definition?

Reply:
From the TDC call this morning: both @handrews @mikekistler agree that it not permitted by the specification and we should fix that in a new major version.

Reply:
ACK